pkg/metrics/statsd/cosmosdb/metrics.go

```diff
@@ -0,0 +1,78 @@
+package cosmodb
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/metrics"
+)
+
+var _ http.RoundTripper = (*tracerRoundTripper)(nil)
+
+type tracerRoundTripper struct {
+	log *logrus.Entry
+	m   metrics.Interface
+	tr  http.RoundTripper
+}
+
+func New(log *logrus.Entry, tr *http.Transport, m metrics.Interface) *tracerRoundTripper {
+	return &tracerRoundTripper{
+		log: log,
+		m:   m,
+		tr:  tr,
+	}
+}
+
+func (t *tracerRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	start := time.Now()
+	resp, err := t.tr.RoundTrip(r)
+	if err != nil {
```

Comment:
If the roundtrip errors, you won't see the metrics.
All the metrics code below should be in a defer above `t.tr.RoundTrip(r)`, then you can just `return t.tr.RoundTrip(r)`.